VotingApplication/VotingApplication.Web/Scripts/Controllers/Create/ConfirmStepController.js

```diff
@@ -0,0 +1,78 @@
+﻿(function () {
+    'use strict';
+
+    angular
+        .module('VoteOn-Create')
+        .controller('ConfirmStepController', ConfirmStepController);
+
+    ConfirmStepController.$inject = ['$scope'];
+
+    function ConfirmStepController($scope) {
+
+        $scope.getNonEmptyChoices = getNonEmptyChoices;
+        $scope.pollHasWarnings = pollHasWarnings;
+        $scope.pollHasErrors = pollHasErrors;
+        $scope.questionIsValid = questionIsValid;
+        $scope.choicesAreValid = choicesAreValid;
+        $scope.styleIsValid = styleIsValid;
+        $scope.expiryIsValid = expiryIsValid;
+        $scope.invitationsAreValid = invitationsAreValid;
+
+        activate();
+
+        function activate() {
+
+        }
+
+        function pollHasWarnings() {
+            return !choicesAreValid() ||
+                   !styleIsValid() ||
+                   !expiryIsValid() ||
+                   !invitationsAreValid();
+        }
+
+        function pollHasErrors() {
+            return !questionIsValid();
+        }
+
+        function questionIsValid() {
+
+            if (!$scope.newPoll.Name) {
+                return false;
+                
+            }
+
+            return true;
+        }
+
+        function choicesAreValid() {
+            if (!getNonEmptyChoices().length && !$scope.newPoll.OptionAdding) {
```

Comment:
`!` can easily be missed. I'd suggest something like
```javascript
if (getNonEmptyChoices().length === 0 && $scope.newPoll.OptionAdding === false) {
```
which makes it a lot easier to read and reason about.